internal/process/cluster.go

```diff
@@ -14,51 +15,57 @@ import (
 func createClusterSamples(config *config.Config) {
 	for _, dc := range config.Datacenters {
 		for _, cluster := range dc.Clusters {
-			// // resolve hypervisor host
-			datacenterName := dc.Datacenter.Name
-
-			//Retrieving the list of host belonging to the cluster
-			hostList := ""
-			for _, hostReference := range cluster.Host {
-				if host, ok := dc.Hosts[hostReference.Reference()]; ok {
-					hostList = hostList + host.Summary.Config.Name + "|"
-				}
-			}
-
-			//Retrieving the list of networks attached to the cluster
-			networkList := ""
-			for _, networkReference := range cluster.Network {
-				if network, ok := dc.Networks[networkReference]; ok {
-					networkList = networkList + network.Name + "|"
-				}
-			}
 
-			//Retrieving the list of datastores attached to the cluster
-			datastoreList := ""
-			for _, datastoreReference := range cluster.Datastore {
-				if datastore, ok := dc.Datastores[datastoreReference]; ok {
-					datastoreList = datastoreList + datastore.Name + "|"
-				}
+			// filtering here will to avoid sending data to backend
+			if config.TagFilteringEnabled() && !config.TagCollector.MatchObjectTags(cluster.Self) {
+				continue
```

Comment:
i would add the debug log like it was in the collector.
`logger.WithField("cluster", cluster.Name).
					Debug("ignoring cluster since no tags matched the configured filters")`